src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,77 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_NUMBER = 5;
+    private static final int MAX_NUMBER = 10;
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        String color = ColorSupplier.getRandomColor();
+        int figureType = random.nextInt(FIGURE_NUMBER);
+
+        switch (figureType) {
+            case 0:
+                return createRandomSquare(color);
+            case 1:
+                return createRandomIsoscelesTrapezoid(color);
+            case 2:
+                return createRandomCircle(color);
+            case 3:
+                return createRandomRightTriangle(color);
+            case 4:
+                return createRandomRectangle(color);
+            default:
+                return createRandomSquare(color);
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        int figureType = random.nextInt(FIGURE_NUMBER);
+
+        switch (figureType) {
+            case 0:
+                return new Square("White", 10);
+            case 1:
+                return new IsoscelesTrapezoid("White", 10, 5, 5);
+            case 2:
+                return new Circle("White", 10);
+            case 3:
+                return new RightTriangle("White", 10, 10);
+            case 4:
+                return new Rectangle("White", 10, 5);
+            default:
+                return new Circle("White", 10);

```

Comment:
https://github.com/mate-academy/jv-oop-advanced/blob/master/checklist.md#all-magic-numbers-in-your-code-should-be-constants